validator_option.go

```diff
@@ -0,0 +1,21 @@
+package jwt
+
+import "time"
+
+// ValidatorOption is used to implement functional-style options that modify the behavior of the parser. To add
+// new options, just create a function (ideally beginning with With or Without) that returns an anonymous function that
+// takes a *ValidatorOptions type as input and manipulates its configuration accordingly.
+type ValidatorOption func(*ValidatorOptions)
+
+// ValidatorOptions represents options that can be used for claims validation
+type ValidatorOptions struct {
```

Comment:
Suggestion: This type should be called `Validator` or rather `validator`. Additionally, I would make this type unexported, at least for now, until our API is more stabilised.